docker/Dockerfile

```diff
@@ -11,4 +11,6 @@ RUN apt-get update \
   && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
 # Requirement, note that v2 or later runs into issues with gem install due to Ruby 2 req
-RUN gem install net-sftp -v 1.1.1    
\ No newline at end of file
+RUN gem install net-sftp -v 1.1.1
+
+RUN chsh -s /bin/bash root
```

Comment:
Was this tested?   From what I can tell, pipeline does *not* respect the configured user's default shell: https://github.com/jenkinsci/durable-task-plugin/commit/3b6eb827aee03048dfe4f2cd9d2aac217f457bcf

Which calls
https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/tasks/Shell.java

I've gone through the sources, and *from what I can tell* Jenkins can set global shells but otherwise always runs /bin/sh rather than whatever your user shell is (within container or not).  

Did I miss something there?